File: test/react-native/features/fixtures/reactnative/scenarios/ContextNativeCustomScenario.kt
package com.reactnative.scenarios

import android.content.Context
import com.bugsnag.android.Bugsnag

class ContextNativeCustomScenario(context: Context): Scenario(context) {

    override fun run() {
        super.run()
        Bugsnag.notify(generateException())
        Thread.sleep(500)
        Bugsnag.setContext("context-native")
        Bugsnag.notify(generateException())
    }
}

File: test/react-native/features/fixtures/scenario-launcher/ios/Scenarios/ContextNativeCustomScenario.m
#import "ContextNativeCustomScenario.h"

@implementation ContextNativeCustomScenario

- (void)run: (RCTPromiseResolveBlock)resolve
     reject:(RCTPromiseRejectBlock)reject {
  NSException *exception = [[NSException alloc] initWithName:@"NSException" reason:@"ContextNativeCustomScenario" userInfo:nil];
  [Bugsnag notify:exception];
  [NSThread sleepForTimeInterval:0.5];
  [Bugsnag setContext:@"context-native"];
  exception = [[NSException alloc] initWithName:@"NSException" reason:@"ContextNativeCustomScenario2" userInfo:nil];
  [Bugsnag notify:exception];
}

@end
